lms/djangoapps/courseware/grades.py

```diff
@@ -509,12 +569,15 @@ def _progress_summary(student, request, course, field_data_cache=None, scores_cl
     max_scores_cache.fetch_from_remote(field_data_cache.scorable_locations)
 
     chapters = []
+    locations_to_children = {}
+    locations_to_weighted_scores = {}
     # Don't include chapters that aren't displayable (e.g. due to error)
     for chapter_module in course_module.get_display_items():
         # Skip if the chapter is hidden
         if chapter_module.hide_from_toc:
             continue
 
+        locations_to_children[chapter_module.location] = []
```

Comment:
You can use a [defaultdict](https://docs.python.org/2/library/collections.html#collections.defaultdict) for situations like this:

```python
locations_to_children = defaultdict(list)
locations_to_children[module_descriptor.parent].append(module_descriptor.location)
```